---
title: "Draft Genome Assemblies of Three Halophilic Bacteria From Soil, Salt Lick, and a Fish Tank"
date: "`r format(Sys.time(), '%d %B %Y')`"
output:
  word_document:
    reference_docx: "paper-files/style_ref.docx"
  pdf_document:
    keep_tex: true
    fig_caption: true
    latex_engine: pdflatex
    fig_width: 7
    fig_height: 6
bibliography: "paper-files/citations.bib"
csl: "paper-files/asm.csl"
header-includes:
  - \usepackage[left]{lineno}
  - \linenumbers
  - \usepackage{setspace}\doublespacing
  - \usepackage{sectsty}\sectionfont{\fontsize{12}{12}\selectfont}
  - \usepackage{sectsty}\subsectionfont{\normalfont\itshape\fontsize{12}{12}\selectfont}
  - \usepackage[round]{natbib}
indent: true
sansfont: Times New Roman
fontsize: 12pt
always_allow_html: true
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = FALSE, eval = TRUE, include = FALSE)
library(data.table)
library(Biostrings)
library(tidyverse)
library(kableExtra)
```

# Abstract:

Halophiles represent an industrially important but still understudied group of organisms. Here, we present a draft genome sequence of three different salt tolerant bacterial strains from the genera Halomonas, Bacillus, and Staphylococcus. These organisms were isolated from soil, a cow salt lick, and a fish tank, and then cultured at 200 g/L NaCl, sequenced, and assembled into draft genome assemblies.

# Background:

Halophiles have been identified as significant sources of stable enzymes that work in extremely high salinity and have a broad range of biotechnological applications [@Liu2019]. There is therefore a pressing need to expand the currently limited genome database of halophilic microorganisms. To help address this need, we cultured three environmental samples and isolated DNA sequences as part of a Course-based Undergraduate Research Experience. From these, we identified three different salt tolerant strains from our assembled draft genomes. Our work can be utilized in studying halophilic mechanisms by providing a broader scope of genomes for future reference [@Becker2014].

# Methods:
## Specimen Collection and Culturing
In order to search for halophilic microbes, we collected 17 samples from a range of environments from around Davis, California, USA. After one week of incubation at 37°C, three samples successfully resulted in bacterial growth in high salt growth medium (200 g/L NaCL). These three samples were sourced in late September or early October 2021 from: 1) soil from the Davis Arboretum, 2) a cow salt lick, and 3) the remains of evaporated water on the glass top of a saltwater fish tank containing a single rabbitfish (*Siganus Guttatas*) held at room temperature (~22°C) under ambient light. To isolate single colonies, we streaked each liquid sample onto high salt agar plates (200g/L NaCl). Plates were incubated for 5-6 days at 37°C. We then picked single colonies from each sample and cultured them in growth medium for one week at 37°C. Samples where then pelleted and DNA was extracted. For liquid culture, Halobacterium 372 (https://www.dsmz.de/microorganisms/medium/pdf/DSMZ_Medium372.pdf) was used. Agar was added to this media to make bacterial growth plates. The soil sample was streaked once, the salt lick sample three times, and the tank sample twice, for a total of six samples.

## Genomic DNA Purification, Sequencing, and Bioinformatics
We purified genomic DNA from each successfully cultured sample using the Monarch DNA extraction kit (New England Biolabs) according to their Quick Protocol. The sample was pelleted and lysed with phosphate buffered saline, lysosomes, tissue lysis buffer, proteinase, and RNAase. DNA was then purified using gDNA binding buffer, a gDNA Purification column elution buffer and wash buffer. To prepare the samples for sequencing, DNA was further purified via ethanol precipitation. DNA sequencing and bioinformatics analysis was performed by MicrobesNG (https://microbesng.com/, see https://microbesng.com/documents/5/MicrobesNG_Methods_Document_-_PDF.pdf for details). Briefly, each sample was lysed, purified, and then sequenced on an Illumina sequencer using 250bp paired-end reads. The resulting reads were trimmed via Trimmomatic 0.30 [@Bolger2014], assembled via SPAdes 3.7 [@Bankevich2012], and annotated using Prokka 1.11 [@Seemann2014].


## Genome Assembly and Quality Assessment
We first used the ContEst16S tool assess contamination in our samples by comparison against known 16S rRNA sequences [@Lee2017]. We considered samples contaminated if the results returned more than one 16S match. Three out of our six samples had contamination and were not further analyzed. We then cleaned and assessed the remaining sequences using the BioStrings R package by filtering out any contigs that were shorter than 500 bp, and then calculated the number of contigs, length of each contig, total genome size, N50, and L50 values, and GC content for the resulting final assemblies [Table 1, @Pages2019]. To determine the possible species identification for each sample, we then ran the largest 16S sequences obtained by the ContEst16S tool through the NCBI’s online BLASTn (Nucleotide Basic Local Alignment Search Tool) tool [@Altschul1990a]. 

<!-- To further assess each genome’s completeness and degree of contamination, we also used the KBase CheckM tool to search for single-copy marker genes (Parks et al. 2015; Arkin et al. 2016). -->

<!-- ## Taxonomy -->
<!-- We obtained the sample taxonomy from Kraken (Wood and Salzburg 2014) by aligning sections of sample genomes to known organism sequences. We calculated Average Nucleotide Identity between each of our assemblies using FastANI on KBase (Jain et al. 2018;Arkin et al. 2016). -->

# Results:

Blast results against our largest 16S sequences from each uncontaminated samples was found to be similar to previously sequenced samples, each with very high percent identity (>99%): the fish tank sample matched to a one to a sample from a known halophilic *Halomonas* species, the Arboretum soil sample to an unidentified *Bacillus* species, and the salt lick sample equally well (identical e-values with 100% query coverage) to many (20+) *Staphylococcus* species (see Table 1). 

# Data Availability:

Assemblies and annotations from all six (contaminated and uncontaminated) samples are available from GenBank with the accession number (will be inserted when available). The rmarkdown script used to analyze the assemblies and produce this manuscript is also available at https://github.com/hemstrow/BIS23_manuscript_2022.

# References:
<div id="refs"></div>

\newpage

# Tables:

```{r import_metadata}
# Import Metadata
meta <- data.table::fread("sample_metadata.txt")
meta <- meta[which(meta$`Reliable Metadata` == "yes"),]
selected_samples <- meta[which(meta$`ContEst16S Decision` == "no"),]
```


```{r stats}
# Re-calculate metrics for transparency

# read in
files <- paste0(selected_samples$`Statistics without reference`, ".fasta")
fastas <- lapply(files, readDNAStringSet)

# function
calc_stats <- function(x){
  # size
  w <- width(x)
  size <- sum(w)
  
  # take only larger contigs
  x <- x[which(w >= 500)]
  w <- w[which(w >= 500)]
  
  # Size > 500
  size_500 <- sum(w)
  
  # N/L50
  lengths <- sort(w, decreasing = TRUE)
  csum <- cumsum(lengths)
  small_enough <- csum >= csum[length(csum)]*.5
  L50 <- min(which(small_enough))
  N50 <- lengths[L50]
  
  # GC
  GC <- alphabetFrequency(x, baseOnly = TRUE, as.prob = FALSE)
  GC <- sum(GC[,c("G", "C")])/size_500
  GC <- GC*100
  
  # number of contigs
  ncont <- length(x)
  
  return(c(`Total Size` = size, `Size >= 500` = size_500, `Number of Contigs` = ncont, `N50` = N50, `L50` = L50, `% GC` = round(GC, 2)))
}

stats <- cbind(`Sample ID` = selected_samples$`Statistics without reference`, 
               `Sample Origin` = selected_samples$`Origins`,
               `Top BLASTn Hit` = selected_samples$`Top Blast Hit`,
               `Accession Number` = selected_samples$Accession,
               `BLASTn % Identity` = selected_samples$`Percent Identity`,
               `BLASTn % Query Coverage` = selected_samples$`Blast Query Cover`,
               `16S Fragment Length` = nchar(selected_samples$`16S Fragment 1`),
               dplyr::bind_rows(lapply(fastas, calc_stats)))

stats <- t(stats)
colnames(stats) <- stats[1,]
stats <- stats[-1,]
stats <- as.data.frame(stats)
stats$` ` <- rownames(stats)
stats <- stats[,c(4,1:3)]

```


```{r, table1, include=TRUE}

flextable::regulartable(stats) %>% flextable::set_caption(., "Sample origin, BLASTn results (Top 16S Hit and Accession Number, % Identity, % Query Coverage, and 16S Fragment Length), total assembled genome size with all contigs, total assembled size only only including contigs larger than 500bp, N50, L50, and % GC for each sample. N50, L50, and % GC are reported for the assembly without contigs smaller than 500bp.")

# kableExtra::landscape(kable(x = stats, format = "latex", linesep ="\\addlinespace", caption = c("Table 1: Total assembled genome size with all contigs and only including contigs larger than 500bp, N50, L50, and % GC for each sample. N50, L50, and % GC are reported for the assembly without contigs smaller than 500bp.")) %>% kable_styling(., latex_options = "scale_down")) #https://haozhu233.github.io/kableExtra/best_practice_for_newline_in_latex_table.pdf
```
```{r, trim_for_ncbi}
meta$file <- paste0(meta$`Statistics without reference`, ".fasta")
for(i in 1:nrow(meta)){
  this_assembly <- readDNAStringSet(meta$file[i])
  this_assembly <- this_assembly[which(width(this_assembly) >= 200)]
  writeXStringSet(this_assembly, paste0(meta$`Statistics without reference`[i], "_clean.fasta"))
}

```


```{r, ncbi}
meta$origin_code <- c("SFT", "SFT", "CSL", "CSL", "CSL", "AS")
meta$strain_name <- paste0(meta$origin_code, c(1, 2, 1, 2, 3, 1))
biosample <- data.frame(sample_name = meta$`Statistics without reference`,
                        organism = paste0(meta$`Most Frequent Genus`, " bacterium "),
                        strain = meta$strain_name,
                        isolation_source = meta$Origins,
                        collection_date = meta$`Collection Date`,
                        geo_loc_name = "USA: California, Davis",
                        sample_type = "cell culture",
                        description = "Cultured at 200g/L NaCl",
                        contamination = paste0(gsub("maybe", "uncertain", meta$`ContEst16S Decision`), ", via to ContEst16S")
                        )

write.table(biosample, "biosample.txt", quote = FALSE, sep = "\t", col.names = TRUE, row.names = FALSE)

genome_information <- data.frame(sample_name = biosample$sample_name,
                                 assembly_method = "SPAdes", 
                                 assembly_method_version = "3.7",
                                 genome_coverage = meta$`Mean Coverage`,
                                 sequencing_technology = "Illumina",
                                 filename = paste0(meta$`Statistics without reference`, "_clean.fasta"))
write.table(genome_information, "genome_information.txt", quote = FALSE, sep = "\t", col.names = TRUE, row.names = FALSE)



```

