---
title: "Draft Genome Assemblies of Three Highly Salt-tolerant Bacteria"
date: "`r format(Sys.time(), '%d %B %Y')`"
output:
  word_document:
    reference_docx: "paper-files/style_ref.docx"
  pdf_document:
    keep_tex: true
    fig_caption: true
    latex_engine: pdflatex
    fig_width: 7
    fig_height: 6
bibliography: "paper-files/citations.bib"
csl: "paper-files/asm.csl"
header-includes:
  - \usepackage[left]{lineno}
  - \linenumbers
  - \usepackage{setspace}\doublespacing
  - \usepackage{sectsty}\sectionfont{\fontsize{12}{12}\selectfont}
  - \usepackage{sectsty}\subsectionfont{\normalfont\itshape\fontsize{12}{12}\selectfont}
  - \usepackage[round]{natbib}
indent: true
sansfont: Times New Roman
fontsize: 12pt
always_allow_html: true
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = FALSE, eval = TRUE, include = FALSE)
library(data.table)
library(Biostrings)
library(tidyverse)
library(kableExtra)
```

# Abstract:

Halophiles represent an industrially important but still understudied group of organisms. Here, we present a draft genome sequence of three different salt tolerant bacterial strains, which were isolated from a range of sources, raised under high salinities, sequenced, and analyzed at the University of California, Davis.

# Background:

Halophiles have been identified as significant sources of stable enzymes that work in extremely high salinity and have a broad range of biotechnological applications [@Liu2019]. There is therefore a pressing need to expand the currently limited genome database of halophilic microorganisms. In response, undergraduate students at the University of California, Davis cultured 3 environmental samples and isolated DNA sequences, and identified three different salt tolerant strains from our assembled draft genomes. Our work can be utilized in studying halophilic mechanisms by providing a broader scope of genomes for future reference [@Becker2014].

# Methods:
## Specimen Collection and Culturing
In order to search for halophilic microbes, we collected 17 samples from a range of environments from around Davis, California. After one week of incubation at 37°C, three samples successfully resulted in bacterial growth in high salt growth medium (200 g/L NaCL). These samples were sourced from: 1) soil from the Davis Arboretum, 2) a cow salt lick, and 3) the remains of evaporated water on the glass top of a salt water fish tank containing a single rabbitfish (*Siganus Guttatas*) held at room temperature (~72) under ambient light. To isolate single colonies, we streaked each liquid sample onto high salt agar plates (200g/L NaCl). Plates were incubated for 5-6 days at 37°C. We then picked single colonies from each sample and cultured them in growth medium for one week at 37°C. Samples where then pelleted and DNA was extracted. For liquid culture, Halobacterium 372 was used. Agar was added to this media to make cacterial growth plates. The soil sample was streaked once, the salt lick sample three times, and the tank sample twice, for a total of six samples.

## Genomic DNA Purification
We purified Genomic DNA from each sample for which we were successfully able to culture colonies using a using the Monarch DNA extraction kit (New England Biolabs) according to their quick protocol. The sample was pelleted and lysed with phosphate buffered saline, lysosomes, tissue lysis buffer, proteinase and RNAase. DNA was then purified using gDNA binding buffer, a gDNA Purification column elution buffer and wash buffer. To prepare the samples for sequencing, DNA was further purified via ethanol precipitation. The DNA was then analyzed and assembled by MicrobesNG (see https://microbesng.com/documents/5/MicrobesNG_Methods_Document_-_PDF.pdf for details). Briefly, each sample was lysed, purified, and the sequenced on an Illumina sequencer using 250bp paired-end reads. The resulting reads were trimmed via Trimmomatic 0.30 [@Bolger2014], assembled via SPAdes 3.7 [@Bankevich2012], and annotated using Prokka 1.11 [@Seemann2014].


## Genome Assembly and Quality Assessment
We cleaned and assessed the sequence produced by MicrobesNG data using the BioStrings R package by filtering out any contigs that were shorter than 500 bp, then calculated the number of contigs, length of each contig, total genome size, N50, and L50 values, and GC content [@Pages2019]. We used the ContEst16S tool to compare our assembled genomes against known 16S rRNA sequences [@Lee2017]. We considered samples as contaminated if the results returned more than one 16S match. We ran the largest 16S sequences obtained by the ContEst16S tool through the NCBI’s online BLASTn (Nucleotide Basic Local Alignment Search Tool) tool to search for potential organisms with matching sequences [@Altschul1990a]. 

<!-- To further assess each genome’s completeness and degree of contamination, we also used the KBase CheckM tool to search for single-copy marker genes (Parks et al. 2015; Arkin et al. 2016). -->

<!-- ## Taxonomy -->
<!-- We obtained the sample taxonomy from Kraken (Wood and Salzburg 2014) by aligning sections of sample genomes to known organism sequences. We calculated Average Nucleotide Identity between each of our assemblies using FastANI on KBase (Jain et al. 2018;Arkin et al. 2016). -->

# Results:

Out of the six samples that we sequenced, three (one from each source) were found to be uncontaminated by ContEst16S. Each of these samples was found to be similar to previously sequenced samples from a distinct genus, each with very high (>99) percent identity; one *Halomonas* species (the fish tank), one *Bacillus* species (the Arboretum soil), and one *Staphylococcus* species (the salt lick; see Table 1). Interestingly, while the tank sample best matched a known halophilic genus (*Halomonas*), the other two did not. However, the *second* best BLAST hit for the soil sample was from a halophilic genus (*Salibacterium*), with a similarly high percent identity (99.73), suggesting that this sample may actually also belong to a known halophilic genus. The cow lick sample shared a percent identity of 100 with many (20+) *Staphlocaccus* samples.

# Data Availability:

Assemblies and annotations for each of our three samples are available from GenBank with the accession number (will be inserted when available). Assemblies and annotations for each of the three other contaminated samples are also available under the same accession. An rmarkdown script used to analyze the assemblies and produce this manuscript is also available at https://github.com/hemstrow/BIS23_manuscript_2022.

# References:
<div id="refs"></div>

\newpage

# Tables:

```{r import_metadata}
# Import Metadata
meta <- data.table::fread("sample_metadata.txt")
meta <- meta[which(meta$`Reliable Metadata` == "yes"),]
selected_samples <- meta[which(meta$`ContEst16S Decision` == "no"),]
```


```{r stats}
# Re-calculate metrics for transparency

# read in
files <- paste0(selected_samples$`Statistics without reference`, ".fasta")
fastas <- lapply(files, readDNAStringSet)

# function
calc_stats <- function(x){
  # size
  w <- width(x)
  size <- sum(w)
  
  # take only larger contigs
  x <- x[which(w >= 500)]
  w <- w[which(w >= 500)]
  
  # Size > 500
  size_500 <- sum(w)
  
  # N/L50
  lengths <- sort(w, decreasing = TRUE)
  csum <- cumsum(lengths)
  small_enough <- csum >= csum[length(csum)]*.5
  L50 <- min(which(small_enough))
  N50 <- lengths[L50]
  
  # GC
  GC <- alphabetFrequency(x, baseOnly = TRUE, as.prob = FALSE)
  GC <- sum(GC[,c("G", "C")])/size_500
  GC <- GC*100
  
  # number of contigs
  ncont <- length(x)
  
  return(c(`Total Size` = size, `Size >= 500` = size_500, `Number of Contigs` = ncont, `N50` = N50, `L50` = L50, `% GC` = round(GC, 2)))
}

stats <- cbind(`Sample ID` = selected_samples$`Statistics without reference`, 
               `Sample Origin` = selected_samples$`Origins`,
               `Top 16S BLASTn Hit` = selected_samples$`Top Blast Hit`,
               `BLASTn % Identity` = selected_samples$`Percent Identity`,
               dplyr::bind_rows(lapply(fastas, calc_stats)))

```


```{r, table1, include=TRUE}

flextable::regulartable(stats) %>% flextable::set_caption(., "Total assembled genome size with all contigs and only including contigs larger than 500bp, N50, L50, and % GC for each sample. N50, L50, and % GC are reported for the assembly without contigs smaller than 500bp.")

# kableExtra::landscape(kable(x = stats, format = "latex", linesep ="\\addlinespace", caption = c("Table 1: Total assembled genome size with all contigs and only including contigs larger than 500bp, N50, L50, and % GC for each sample. N50, L50, and % GC are reported for the assembly without contigs smaller than 500bp.")) %>% kable_styling(., latex_options = "scale_down")) #https://haozhu233.github.io/kableExtra/best_practice_for_newline_in_latex_table.pdf
```

